isaacgym_utils/assets/franka.py

```diff
@@ -223,7 +230,86 @@ def set_attractor_props(self, env_idx, name, props):
 
         self._scene.gym.set_attractor_properties(env_ptr, ath, attractor_props)
 
-    def set_ee_transform(self, env_idx, name, transform):
+    def set_ee_transform(self, env_idx, name, ee_transform, **set_ee_kwargs):
+
+        # TODO: deal with ee offsets
+
+        ik_solution_found, ik_joint_angles_wo_gripper = self.joints_from_ee_transform_ik(
+            env_idx,
+            name,
+            ee_transform,
+            **set_ee_kwargs,
+        )
+
+        if ik_solution_found:
+            # Set targets for joints/ee etc
+            gripper_angles = self.get_joints(env_idx, name)[7:].astype("float")
+            new_joint_angles = np.concatenate(
+                (
+                    ik_joint_angles_wo_gripper,
+                    gripper_angles,
+                )
+            )
+
+            # This should "snap" the robot to the block
+            self.set_joints(env_idx, name, new_joint_angles)
+            self.set_joints_targets(env_idx, name, new_joint_angles)
+            self.set_ee_transform_target(env_idx, name, ee_transform)
+
+        return ik_solution_found
+
+    def joints_from_ee_transform_ik(self, env_idx, name, ee_transform, ik_robot_joints_hint='default'):
+
+        if isinstance(ik_robot_joints_hint, str) and ik_robot_joints_hint == 'default':
+            # this was found by setting ee position to [0.4, 0., 0.8] and same initial ee rot
+            # this helps IK converge to a sensible solution for most workspace EE poses
+            robot_joints_for_ik = np.array([
+                1.1024622e-02,
+                -4.0491045e-01,
+                -1.0826388e-02,
+                -2.5944960e00,
+                -5.4867277e-03,
+                2.1895969e00,
+                7.8967488e-01,
+            ])
+        elif isinstance(ik_robot_joints_hint, str) and ik_robot_joints_hint == 'current':
+            robot_joints_for_ik = self.get_joints(env_idx, name)[:7].astype(
+                "float"
+            )
+        else:
+            robot_joints_for_ik = ik_robot_joints_hint
+
+        # where is the franka base
+        base_tform = self.get_base_transform(env_idx, name)
+        desired_ee_pos = ee_transform.p
+
+        desired_ee_pos_wrt_base = vec3_to_np(desired_ee_pos - base_tform.p)
+
+        quat_desired = UnitQuaternion(
```

Comment:
replace spatialmath w/ a RigidTransform:

```python
T = RigidTransform(translation=desired_ee_pos_wrt_base, rotation=quat_to_np(ee_transform.r, 'wxyz')).matrix
```

need ot import `quat_to_np` from `math_utils`

Reply:
The reason I didn't use `RigidTransform` initially is because `ikine_LM` requires this object to be of type `SE3` from `spatialmath`, so we need to keep `spatial math`
https://github.com/petercorke/robotics-toolbox-python/blob/51aa8bbb3663a7c815f9880d538d61e7c85bc470/roboticstoolbox/robot/IK.py#L128